omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+            print("Robot is near joint limits! No trajectory will be computed")
+            return None
+
+        # Define the plan config
+        plan_cfg = lazy.curobo.wrap.reacher.motion_gen.MotionGenPlanConfig(
+            enable_graph=False,
+            # enable_graph_attempt=max_attempts,
+            max_attempts=max_attempts,
+            timeout=2.0,
+            ik_fail_return=5,
+            # num_ik_seeds=10,
+            # num_trajopt_seeds=10,
+            enable_finetune_trajopt=enable_finetune_trajopt,
+            finetune_attempts=1,
+            success_ratio=1.0 / self.batch_size if success_ratio is None else success_ratio,
+        )
+
+        # Refresh the collision state
+        self.update_obstacles()
+
+        # self.mg.reset()
+
+        # Check if batch size is greater than 1, and if so, make sure target pos / quat are the appropriate shapes
+        if len(target_pos.shape) == 1:
+            target_pos = th.tile(target_pos.view(1, 3), (self.batch_size, 1))
+        if len(target_quat.shape) == 1:
+            target_quat = th.tile(target_quat.view(1, 4), (self.batch_size, 1))
+        assert target_pos.shape[0] == self.batch_size
+        assert target_quat.shape[0] == self.batch_size
+
+        # Make sure the specified target pose is in the robot frame
+        robot_pos, robot_quat = self.robot.get_position_orientation()
+        if not is_local:
+            target_pose = th.zeros((self.batch_size, 4, 4))
+            target_pose[:, 3, 3] = 1.0
+            target_pose[:, :3, :3] = T.quat2mat(target_quat)
+            target_pose[:, :3, 3] = target_pos
+            inv_robot_pose = th.eye(4)
+            inv_robot_ori = T.quat2mat(robot_quat).T
+            inv_robot_pose[:3, :3] = inv_robot_ori
+            inv_robot_pose[:3, 3] = -inv_robot_ori @ robot_pos
+            target_pose = inv_robot_pose.view(1, 4, 4) @ target_pose
+            target_pos = target_pose[:, :3, 3]
+            target_quat = T.mat2quat(target_pose[:, :3, :3])
+
+        # Map xyzw -> wxyz quat
+        target_quat = target_quat[:, [3, 0, 1, 2]]
+
+        # Create IK goal
+        ik_goal_batch = lazy.curobo.types.math.Pose(
+            position=self._tensor_args.to_device(target_pos).contiguous(),
+            quaternion=self._tensor_args.to_device(target_quat).contiguous(),
+        )
+
+        # Construct initial state
+        q_pos = th.stack([self.robot.get_joint_positions()] * self.batch_size, axis=0)
+        q_vel = th.stack([self.robot.get_joint_velocities()] * self.batch_size, axis=0)
+        q_eff = th.stack([self.robot.get_joint_efforts()] * self.batch_size, axis=0)
+        sim_js_names = list(self.robot.joints.keys())
+        cu_js_batch = lazy.curobo.types.state.JointState(
+            position=self._tensor_args.to_device(q_pos),
+            velocity=self._tensor_args.to_device(q_vel) * 0.0,
+            # TODO: Ideally this should be nonzero, but curobo fails to compute a solution if so
```

Comment:
sorry what do you mean by this? You want the result to have 0 velocity and 0 acceleration?

Reply:
and 0 jerk too, above all 3 lines.

Reply:
done